immuneML/ml_methods/PWM.py

```diff
@@ -0,0 +1,174 @@
+import csv
+import yaml
+import datetime
+
+import numpy as np
+import pandas as pd
+
+from pathlib import Path
+from immuneML.ml_methods.GenerativeModel import GenerativeModel
+from scripts.specification_util import update_docs_per_mapping
+from immuneML.util.PathBuilder import PathBuilder
+
+class PWM(GenerativeModel):
+
+    def get_classes(self) -> list:
+        pass
+
+    def __init__(self, parameter_grid: dict = None, parameters: dict = None):
+        parameters = parameters if parameters is not None else {}
+        parameter_grid = parameter_grid if parameter_grid is not None else {}
+        self._alphabet = ""
+        self.generated_sequences = []
+        super(PWM, self).__init__(parameter_grid=parameter_grid, parameters=parameters)
+
+    def _get_ml_model(self, cores_for_training: int = 2, dataset=None):
+
+
+        sequences = dataset.split(" ")
```

Comment:
the same function should not do different types of things across subclasses -> if get_ml_model returns an untrained object of the appropriate class, then none of the subclasses should return the trained object here. Also this part of the code comes from sklearn examples, but your classes here don't follow the same pattern (and they shouldn't), so no need to use the functions that existed there and that are not meaningful here..